test/rpc-server/runTests.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+from jsonrpcclient import request, request_uuid, notification
+import json, socket, os, subprocess, difflib
+
+HOST = "127.0.0.1"  # Standard loopback interface address (localhost)
+PORT = 31337  # Port to listen on (non-privileged ports are > 1023)
+CREATE_MISSING_GOLDEN = os.getenv("CREATE_MISSING_GOLDEN", 'False').lower() in ('true', '1', 't')
+RECREATE_BROKEN_GOLDEN = os.getenv("RECREATE_BROKEN_GOLDEN", 'False').lower() in ('true', '1', 't')
+
+
+def recv_all(sock):
+    BUFF_SIZE = 4096 # 4 KiB
+    data = b''
+    while True:
+        part = sock.recv(BUFF_SIZE)
+        data += part
+        if len(part) < BUFF_SIZE:
+            # either 0 or end of data
```

Comment:
This might not be a good solution: https://stackoverflow.com/a/7337584.

Did you consider implementing the server using HTTP? Alternatively, if raw sockets are to be used then perhaps  the client can read the socket until it successfully parses the JSON (or runs into an error), but that's harder to implement.

Reply:
This is just a test script, the pyk implementation should obviously make sure these things are correct. Using HTTP seems like an overkill given the RPC server is meant to be used locally only, as far as I understand. I think the vscode RPC server also just uses raw sockets without HTTP.

Reply:
I'm not against using TCP without HTTP, but in that case it would be much easier to implement clients if the server defined a super thin application layer at least, e.g.:

- The JSON payload is written in a single line, terminated by newline (my preference); or
- The first fixed number of bytes is an integer indicating the size of the rest of the payload in bytes.

Otherwise, we probably won't be able to use the JSON parser in the Python standard library (but maybe we could us [ijson](https://github.com/ICRAR/ijson) instead).

Reply:
The server does indeed send a single line terminated by a newline (or at least should!)

Reply:
That's great, it simplifies things a lot. Please document it in `JSON-RPC-Server-API.md` so that it becomes part of the contract.

Reply:
added to the doc now